src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TransPatterns.java

```diff
@@ -524,23 +528,59 @@ private void handleSwitch(JCTree tree,
                                               .filter(l -> !TreeInfo.isNullCaseLabel(l))
                                               .collect(List.collector());
                 }
-                if (clearedPatterns.size() == 1 && clearedPatterns.head.hasTag(Tag.PATTERNCASELABEL) && !previousCompletesNormally) {
-                    JCPatternCaseLabel label = (JCPatternCaseLabel) clearedPatterns.head;
+
+                boolean validCaseLabelList;
+                if (clearedPatterns.size() > 1) {
+                    validCaseLabelList = clearedPatterns.stream().allMatch(cP -> cP.hasTag(Tag.PATTERNCASELABEL));
+                } else validCaseLabelList = clearedPatterns.size() == 1 && clearedPatterns.head.hasTag(Tag.PATTERNCASELABEL);
```

Comment:
style: I would use `{}` for the code in the `else` branch here same as for the `then` branch